gxa/src/main/java/uk/ac/ebi/atlas/controllers/rest/DifferentialJsonResultsParser.java

```diff
@@ -31,63 +32,69 @@ public DifferentialJsonResultsParser(ContrastTrader contrastTrader)
         this.contrastTrader = contrastTrader;
     }
 
-    public ImmutableList<DiffAnalytics> parseDifferentialResults(JsonObject obj)
-    {
-        try
-        {
+    public ImmutableList<DiffAnalytics> parseDifferentialResults(JsonObject obj) {
+
+        try {
             ImmutableList.Builder<DiffAnalytics> diffAnalyticsListBuilder = ImmutableList.builder();
             JsonArray results = obj.getAsJsonArray("results");
 
             for(JsonElement e : results) {
 
-                DiffAnalytics diffAnalytics = buildDiffAnalyticObject(e);
+                DiffAnalytics diffAnalytics = buildDiffAnalyticsObject(e);
 
-                if(diffAnalytics != null)
-                {
+                if(Optional.ofNullable(diffAnalytics).isPresent()) {
                     diffAnalyticsListBuilder.add(diffAnalytics);
                 }
             }
 
             return diffAnalyticsListBuilder.build();
         }
-        catch (RuntimeException e)
-        {
-            throw new RuntimeException("Input Json is missing a results field OR results field is not an Array",e);
+        catch (IllegalFormatException e) {
```

Comment:
I don’t think this is the exception you want. From https://docs.oracle.com/javase/8/docs/api/java/util/IllegalFormatException.html:
> Unchecked exception thrown when a format string...

A format string is what’s used by `String::format`. How about a [JsonParseException](https://google.github.io/gson/apidocs/com/google/gson/JsonParseException.html)?